core/include/gnuradio-4.0/DataSet.hpp

```diff
@@ -108,8 +108,8 @@ static_assert(DataSetLike<DataSet<float>>, "DataSet<float> concept conformity");
 static_assert(DataSetLike<DataSet<double>>, "DataSet<double> concept conformity");
 
 // public type definitions to allow simple reflection
-using DataSet_float  = DataSet<double>;
-using DataSet_double = DataSet<float>;
+using DataSet_double = DataSet<double>;
```

Comment:
Let's get rid of this entirely since this not used in the lib.

Reply:
It's not easily possible to remove these, as these are registered as `ENABLE_REFLECTION(gr::DataSet_float, ...)`

That way when registering a block with T=DataSet_float, one can pass "DataSet_float" to the PluginLoader and the right handler is found to instantiate the block. With `ENABLE_REFLECTION_FOR_TEMPLATE(gr::DataSet)`, the type name used is `gr::DataSet<Ts...>` (that's what the meta::type_name returns for reflectable templates), so there's no handler for e.g. "gr::DataSet<float>" found.

Reply:
```cpp
ENABLE_REFLECTION_FOR_TEMPLATE(gr::DataSet, timestamp, axis_names, axis_units, axis_values, extents, layout, signal_names, signal_units, signal_values, signal_errors, signal_ranges, meta_information, timing_events)
```
... is your friend.

Reply:
That's what i tried, but that gives `gr::DataSet<Ts...>` as name. This will also hit us when adding type support to load_grc()/ save_grc(), maybe we should solve then.